src/main/java/org/commcare/cases/entity/AsyncEntity.java

```diff
@@ -169,15 +169,13 @@ public String getSortField(int i) {
                                 sortData[i] = "<invalid xpath: " + xpe.getMessage() + ">";
                             }
                         }
-                        return sortData[i];
                     }
-                    return sortData[i];
                 }
             }
         } catch (IOException e) {
             Logger.exception("Error while getting sort field", e);
         }
-        return null;
+        return sortData[i];

```

Comment:
We [cache the properties](https://github.com/dimagi/commcare-android/blob/d827345844da59537e2afc61cbaa67ab46db5395/app/src/org/commcare/models/database/user/models/CommCareEntityStorageCache.java#L266) when we load the case list so the default value is used [here](https://github.com/dimagi/commcare-core/blob/f4d0df4f526b6b097fdd313c6489fd902562246e/src/main/java/org/commcare/cases/entity/AsyncEntity.java#L138C21-L138C40) when `sortData[i] != null`

Reply:
I'm not super comfortable with this change as currently documented. 

The PR description mentions that the null condition was introduced in error, but doesn't reference the change or clarify why the change was wrong, and it doesn't spell out why the method is currently falling through to the null condition and why `sortData[i]` is a safe fallback in all cases. 

There's still at least one `null` return condition above on line 142, so if this method should never return null, this change also isn't really sufficient. In that case `sortData[i]` is **also** explicitly null, so it seems problematic to presume that this will ~always result in a safe return.

Reply:
@ctsims I added more details to the PR description, let me know if it's clearer now. The _null_ on line 142 seems right, when there is no data, we don't need to continue the execution.

Reply:
@avazirna Looking at the previous version of this class it seems like we should be returning `null` here (in case of an IOException encountered) and instead should be returning `sortData[i]` after the closing of ` if (sortData[i] == null) { }` block on line 174.

Reply:
@shubham1g5 `sortData` is initialised with `null` values so in case an IOException is thrown, the current approach should return the element in `sortData`, which can be `null` or an existing value. For readability sake, I'm happy to follow your lead.

Reply:
[Refactor](https://github.com/dimagi/commcare-core/pull/1439/commits/9e50d74ef0864b1002b803ebb585bdf4b01651c1)